src/utils/logging.ts

```diff
@@ -1,33 +1,60 @@
+import {createLogger, format, transports, transport, Logger} from 'winston'
+
 import {ContextualizedControl, contextualizeEvaluation, ExecJSON} from 'inspecjs'
-import winston from 'winston'
+/**
+ * Summary type represents a summary of an HDF execution.
+ * @property {string[]} profileNames - An array of profile names.
+ * @property {number} controlCount - The total number of controls.
+ * @property {number} passedCount - The number of controls that passed.
+ * @property {number} failedCount - The number of controls that failed.
+ * @property {number} notApplicableCount - The number of controls that are not applicable.
+ * @property {number} notReviewedCount - The number of controls that were not reviewed.
+ * @property {number} errorCount - The number of controls that resulted in an error.
+ */
 
 export type Summary = {
-    profileNames: string[];
-    controlCount: number;
-    passedCount: number;
-    failedCount: number;
-    notApplicableCount: number;
-    notReviewedCount: number;
-    errorCount: number;
+  profileNames: string[];
+  controlCount: number;
+  passedCount: number;
+  failedCount: number;
+  notApplicableCount: number;
+  notReviewedCount: number;
+  errorCount: number;
 }
+/**
+ * createWinstonLogger function creates a Winston logger.
+ * @param {string} mapperName - The name of the mapper.
+ * @param {string} [level='info'] - The log level. Default is 'info'.
+ * @returns {Logger} A Winston logger.
+ */
+
+export function createWinstonLogger(mapperName: string, level = 'info'): Logger {
+  const transportList: transport[] = [
+    new transports.File({filename: 'saf-cli.log'}),
```

Comment:
Where does this get stored? I agree with @Amndeep7 that if we want to change the default, we should make sure to document where it's going and where to find it. Maybe we leave the default as outputting to the console and update Winston logging in an overall refactor ticket after this PR?

Reply:
This will be 'fixed' when we do the logging PR. At the moment this is just a bit of cruft. I will see if I can work on it tonight to move this around but if not, we can merge and address it directly after.

Reply:
technically it goes in the $PWD of where you exec'd the command which isn't correct - it should - default to either the defined $SAF_CLI_LOG env var first, then to `~/.saf-cli/` second in a big boy application.

We can add that as a point on the logger PR

Reply:
Issue opened for this. We will open a PR for logging in the next itterations.